#166265853 Implement view and delete books

chuxmykel opened the pull request:
#### What does this PR do?
This PR adds the 'view specific book' and 'delete book' functionality to the application.

#### Description of task to be completed
- user should be able to fetch all books
- user should be able to fetch a specific book
- user should be able to delete a specific book
- work on password reset validation

#### How should this be manually tested
- send a `GET` request to `/books/<id>`.
- send a `DELETE` request to `/books/<id>`.

#### Any background context you want to provide
- None

#### Relevant Pivotal Tracker stories
- [#166265853](https://www.pivotaltracker.com/n/projects/2351121/stories/166265853)

#### Screenshots
![Screenshot from 2019-06-07 08-27-46](https://user-images.githubusercontent.com/44305641/59088017-398b1e00-88fe-11e9-9ce3-ebb3697b0e0a.png)
![Screenshot from 2019-06-07 08-26-55](https://user-images.githubusercontent.com/44305641/59088020-3a23b480-88fe-11e9-9bd5-f87b210ac208.png)
![Screenshot from 2019-06-07 07-45-55](https://user-images.githubusercontent.com/44305641/59088021-3a23b480-88fe-11e9-875d-65a1a14ff75a.png)
![Screenshot from 2019-06-07 07-32-01](https://user-images.githubusercontent.com/44305641/59088022-3abc4b00-88fe-11e9-9ce6-cf2f0fdba2ff.png)

chuxmykel committed:
feat(crud): view/delete books

- work on view specific book endpoint
- work on delete book endpoint
- validate and test endpoints
- validate password reset endpoint

[Delivers #166265853]

chuxmykel committed:
feat(password reset): display relevent messages

- display error and success messages on password reset UI

[Delivers #166265853]

chuxmykel committed:
Merge branch 'develop' into ft/166265853/view-delete-books

emp-daisy commented on .env.sample:
```diff
@@ -1,4 +1,4 @@
-PORT = an integer representing the port through which the server will be accessed e.g. 3000
+ORT = an integer representing the port through which the server will be accessed e.g. 3000
```
Hey! What's the change here?

Hadeneekeh replied:
Okay

chuxmykel replied:
Oh Boy. Error. Sorry. I'll fix that now

Hadeneekeh commented:
Hello Chux, well done. Since password reset validation is entirely different from viewing and deleting a book, I think it should not be included on this branch and subsequently on this PR.

emp-daisy commented:
I asked him to include it there @Hadeneekeh

encodedBicoding reviewed (dismissed):
Fix change in .env.sample file

Hadeneekeh commented:
> I asked him to include it there @Hadeneekeh

Okay

chuxmykel committed:
feat(docs): document feature

- write documentation for view and delete endpoints

[Delivers #166265853]

chuxmykel committed:
Merge branch 'ft/166265853/view-delete-books' of github.com:emp-daisy/HelloBooksLib into ft/166265853/view-delete-books

chuxmykel committed:
feat(docs): document feature

- write documentation for view and delete endpoints
- edit .env.sample to correct mistake

[Delivers #166265853]
